front/app/components/IdeaCard/utils.ts

```diff
@@ -0,0 +1,102 @@
+import { IMAGES_LOADED_EVENT } from 'components/admin/ContentBuilder/constants';
+
+import { removeSearchParams } from 'utils/cl-router/removeSearchParams';
+import eventEmitter from 'utils/eventEmitter';
+import { scrollToElement } from 'utils/scroll';
+
+const SCROLL_OFFSETS = {
+  MOBILE: 150,
+  DESKTOP: 300,
+} as const;
+
+/*
+ * handleScrollToCard:
+ * Handles the scroll-to-card with project description image loading considerations.
+ */
+export const handleScrollToCard = (
+  cardId: string,
+  smallerThanPhone: boolean
+): (() => void) => {
+  // Define the scroll function
+  const performScroll = () =>
+    scrollToCardAndCleanUpUrl(cardId, smallerThanPhone);
+
+  // Get the project description element
+  const projectDescription = document.querySelector(
+    '[id^="project-description"]'
+  );
+
+  // Check if project description images are loaded,
+  // if so, perform scroll immediately.
+  if (projectDescription && checkImagesLoaded(projectDescription)) {
+    return () => {
```

Comment:
Won't this only run on clean-up since it is being used in an effect? I suspect it will. It might be worth checking
Maybe
```
if (projectDescription && checkImagesLoaded(projectDescription)) {
  performScroll();  // Execute immediately
  return () => {};  // Return empty cleanup
}
```